### PR TITLE
Misc fix for feature-serverside-apply

### PR DESCRIPTION
**Pull request by tedyu**
**What type of PR is this?**
/kind bug

**What this PR does / why we need it**:
This PR tries to fix some issues found in feature-serverside-apply.
For gvkparser.go, modelName is included in error message

**Special notes for your reviewer**:

```release-note
NONE
```

### CODE REVIEW

**Comment by tedyu** [comment]
cc @apelisse @jennybuckley

File: staging/src/k8s.io/apiserver/pkg/endpoints/handlers/fieldmanager/internal/gvkparser.go [hunk 1]
```diff
@@ -55,11 +55,15 @@ func newGVKParser(models proto.Models) (*gvkParser, error) {
...
 		if model == nil {
-			panic("ListModels returns a model that can't be looked-up.")
+			panic(fmt.Sprintf("ListModels returns a model that can't be looked-up for: %v", modelName))
 		}
 		gvkList := parseGroupVersionKind(model)
 		for _, gvk := range gvkList {
 			if len(gvk.Kind) > 0 {
+	                        _, ok := parser.gvks[gvk]
+                                if ok {
+                                    return nil, fmt.Errorf("Duplicate entry for %v", gvk)
```
Comment (apelisse) [inline staging/src/k8s.io/apiserver/pkg/endpoints/handlers/fieldmanager/internal/gvkparser.go]: Sorry I forgot to mention, you shouldn't capitalize returned errors (https://github.com/golang/go/wiki/CodeReviewComments#error-strings)

File: staging/src/k8s.io/apiserver/pkg/endpoints/handlers/fieldmanager/internal/gvkparser.go [hunk 2]
```diff
@@ -55,11 +55,15 @@ func newGVKParser(models proto.Models) (*gvkParser, error) {
 	for _, modelName := range models.ListModels() {
 		model := models.LookupModel(modelName)
 		if model == nil {
-			panic("ListModels returns a model that can't be looked-up.")
+			panic(fmt.Sprintf("ListModels returns a model that can't be looked-up for: %v", modelName))
```
Comment (kwiesmueller) [inline staging/src/k8s.io/apiserver/pkg/endpoints/handlers/fieldmanager/internal/gvkparser.go]: Do we want to keep the panic here as well?
cc. @apelisse / @jennybuckley

Comment (apelisse) [reply staging/src/k8s.io/apiserver/pkg/endpoints/handlers/fieldmanager/internal/gvkparser.go]: Yeah, this is not an error, this is an assertion.

**Comment by kwiesmueller** [comment]
/assign @lavalamp 
We are not allowed to approve here

**Comment by tedyu** [comment]
Ping @lavalamp

**Comment by lavalamp** [comment]
/lgtm
/approve

(github notifications are not a reliable way of getting my attention, sorry for delay)

**Pull request merged**